src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -2029,7 +2136,12 @@ private TokenFlags ScanStringExpandable(StringBuilder sb, StringBuilder formatSb
                     if (c1 != 0)
                     {
                         SkipChar();
-                        c = Backtick(c1);
+                        string str = Backtick(c1);
+                        if (IsSurrogatePair(str, ref c, sb, formatSb))
+                        {
+                            // The character has been processed and appended to the string builders
```

Comment:
@lzybkr I've been looking at the C# implementation for string literal lexing, we could follow their approach.  The code changes to roughly this:
```csharp
        private char Backtick(char c, out char surrogateCharacter)
        {
            surrogateCharacter = s_invaldChar;

            switch (c)
            {
                case '0': return '\0';
                case 'a': return '\a';
                case 'b': return '\b';
                case 'e': return '\u001b';
                case 'f': return '\f';
                case 'n': return '\n';
                case 'r': return '\r';
                case 't': return '\t';
                case 'u': return ScanUnicodeEscape(out surrogateCharacter);
                case 'v': return '\v';
                default: return c;
            }
        }

        private char ScanUnicodeEscape(out char surrogateCharacter)
        {
            ...
         }

        // NOTE: this is lifted directly from the OSS C# lexer implementation
        private static char GetCharsFromUtf32(uint codepoint, out char lowSurrogate)
        {
            if (codepoint < (uint)0x00010000)
            {
                lowSurrogate = s_invaldChar;
                return (char)codepoint;
            }
            else
            {
                Diagnostics.Assert(codepoint > 0x0000FFFF && codepoint <= 0x0010FFFF, "Codepoint is outside range for surrogate pair");
                Debug.Assert(codepoint > 0x0000FFFF && codepoint <= 0x0010FFFF);
                lowSurrogate = (char)((codepoint - 0x00010000) % 0x0400 + 0xDC00);
                return (char)((codepoint - 0x00010000) / 0x0400 + 0xD800);
            }
        }
```
The `IsSurrogateChar()` method goes away and the resulting `Backtick` call sites look like this:
```csharp
                else if (c == '`')
                {
                    // If end of input, go ahead and append the backtick and issue an error later
                    char c1 = PeekChar();
                    if (c1 != 0)
                    {
                        SkipChar();
                        c = Backtick(c1, out char c2);
                        if (c2 != s_invaldChar)
                        {
                            sb.Append(c).Append(c2);
                            formatSb.Append(c).Append(c2);
                            continue;
                        }
                    }
                }
```
What do you think? Is it worth making this change?  It is a bit weird that they have their own GetCharsFromUtf32() but I figure it's a solid implementation.  :-)